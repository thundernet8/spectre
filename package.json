{
    "name": "spectre",
    "version": "0.0.1",
    "description": "React implemetation of Spectre.css",
    "main": "index.js",
    "repository": "git@github.com:thundernet8/spectre.git",
    "author": "wuxueqian <wuxueqian2010@hotmail.com>",
    "license": "MIT",
    "devDependencies": {
        "@types/enzyme": "^2.8.2",
        "@types/jest": "^20.0.2",
        "@types/react": "^15.0.37",
        "@types/react-dom": "^15.5.1",
        "babel-core": "^6.25.0",
        "babel-eslint": "^7.2.3",
        "babel-jest": "^20.0.3",
        "babel-loader": "^7.1.1",
        "babel-plugin-add-module-exports": "^0.2.1",
        "babel-plugin-import": "^1.2.1",
        "babel-plugin-transform-es2015-modules-umd": "^6.24.1",
        "babel-plugin-transform-runtime": "^6.23.0",
        "babel-preset-es2015": "^6.24.1",
        "babel-preset-jest": "^20.0.3",
        "babel-preset-react": "^6.24.1",
        "babel-preset-stage-0": "^6.24.1",
        "babel-register": "^6.24.1",
        "cross-env": "^5.0.1",
        "css-loader": "^0.28.4",
        "cssnano": "^3.10.0",
        "enzyme": "^2.9.1",
        "enzyme-to-json": "^1.5.1",
        "eslint": "^4.2.0",
        "eslint-config-airbnb": "^15.0.2",
        "eslint-plugin-babel": "^4.1.1",
        "eslint-plugin-import": "^2.7.0",
        "eslint-plugin-jsx-a11y": "^6.0.2",
        "eslint-plugin-markdown": "^1.0.0-beta.6",
        "eslint-plugin-react": "^7.1.0",
        "extract-text-webpack-plugin": "^3.0.0",
        "hierarchical-copy": "^0.3.2",
        "jest": "^20.0.4",
        "jest-css-modules": "^1.1.0",
        "less": "^2.7.2",
        "less-loader": "^4.0.5",
        "lint-staged": "^4.0.1",
        "mkdirp": "^0.5.1",
        "optimize-css-assets-webpack-plugin": "^2.0.0",
        "pre-commit": "^1.2.2",
        "react-test-renderer": "^15.6.1",
        "rimraf": "^2.6.1",
        "style-loader": "^0.18.2",
        "stylelint": "^7.13.0",
        "stylelint-config-standard": "^16.0.0",
        "ts-jest": "^20.0.7",
        "ts-loader": "^2.2.2",
        "tslint": "^5.5.0",
        "tslint-eslint-rules": "^4.1.1",
        "tslint-react": "^3.0.0",
        "typescript": "^2.4.1",
        "typescript-babel-jest": "^1.0.4",
        "webpack": "^3.2.0"
    },
    "scripts": {
        "test": "jest --config .jest.js",
        "dist": "rimraf dist && mkdirp dist && cross-env NODE_ENV=production webpack --hide-modules --colors --config webpack.config.babel.js",
        "lib": "rimraf lib && tsc && rcp './components/**/*.less' ./lib",
        "prepublish": "npm run dist && npm run lib && npm run lint",
        "lint": "npm run lint:ts && npm run lint:es && npm run lint:style",
        "lint:ts": "tsc && tslint -e node_modules typings -c .tslintrc.json components/**/*.tsx",
        "lint:es": "eslint tests components site/src ./webpack.config.babel.js --ext '.js,.jsx'",
        "lint:style": "stylelint \"{site/src,components}/**/*.less\" --syntax less",
        "lint-staged": "lint-staged",
        "lint-staged:ts": "tsc && tslint --fix -c .tslintrc.json",
        "lint-staged:es": "eslint ./webpack.config.babel.js",
        "lint-staged:md": "eslint --ext '.md'",
        "site": "npm run lib && cd site && npm run dev"
    },
    "lint-staged": {
        "components/**/*.tsx": ["lint-staged:ts"],
        "{tests,components}/**/*.{js,jsx}": ["lint-staged:es"],
        "{components,site}/**/*.less": "stylelint --syntax less"
    },
    "pre-commit": "lint-staged && git add -A",
    "dependencies": {
        "classnames": "^2.2.5",
        "omit.js": "^1.0.0",
        "prop-types": "^15.5.10",
        "react": "^15.6.1",
        "react-dom": "^15.6.1"
    }
}
