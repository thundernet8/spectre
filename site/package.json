{
  "name": "site",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "devDependencies": {
    "webpack": "^3.2.0"
  }
}
