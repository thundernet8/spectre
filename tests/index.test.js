import { version } from "../package.json";

describe("dist files", () => {
    it("export modules correctly", () => {
        const spt = process.env.CI
            ? require("../dist/spectre") // eslint-disable-line global-require
            : require("../components"); // eslint-disable-line global-require
        expect(Object.keys(spt)).toMatchSnapshot();
    });

    it("have matched version", () => {
        const spt = require("../dist/spectre"); // eslint-disable-line global-require
        expect(spt.version).toBe(version);
    });
});
