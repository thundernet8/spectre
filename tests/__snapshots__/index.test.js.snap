// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dist files export modules correctly 1`] = `
Array [
  "Button",
  "Col",
  "Row",
  "version",
]
`;
